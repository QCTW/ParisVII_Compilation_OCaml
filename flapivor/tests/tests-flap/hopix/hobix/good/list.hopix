type list = Nil | Cons (int, list)

fun range (i, j) = if i > j then Nil else Cons (i, range (i + 1, j))

fun len (l) = l ? Nil => 0 | Cons (_, l) => 1 + len (l)

val x = len (range (1, 100))
