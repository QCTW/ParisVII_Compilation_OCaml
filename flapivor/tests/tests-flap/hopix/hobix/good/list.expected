range = <fun>
len = <fun>
x = 100
